src/main/java/com/northconcepts/datapipeline/examples/cookbook/ReadAnXmlFileWithByteOrderMark.java

```diff
@@ -0,0 +1,50 @@
+package com.northconcepts.datapipeline.examples.cookbook;
+
+import java.io.File;
+import java.io.FileInputStream;
+
+import javax.xml.stream.XMLStreamReader;
+
+import org.apache.commons.io.ByteOrderMark;
+import org.apache.commons.io.input.BOMInputStream;
+import org.apache.log4j.Logger;
+
+import com.northconcepts.datapipeline.core.DataEndpoint;
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.core.Record;
+import com.northconcepts.datapipeline.internal.lang.Util;
+import com.northconcepts.datapipeline.internal.xpath.XmlNodeReader;
+import com.northconcepts.datapipeline.xml.XmlReader;
+
+public class ReadAnXmlFileWithByteOrderMark {
+    
+    public static final Logger log = DataEndpoint.log;
+
+    public static void main(String[] args) throws Throwable {
+        BOMInputStream bomInputStream = BOMInputStream.builder()
+                .setInputStream(new FileInputStream(new File("example/data/input/xml_with_byte_order_marks.xml")))
+                .setByteOrderMarks(ByteOrderMark.UTF_8, ByteOrderMark.UTF_16BE, ByteOrderMark.UTF_16LE,
+                        ByteOrderMark.UTF_32BE, ByteOrderMark.UTF_32LE)
+                .setInclude(false)
+                .get();
+        
+        XMLStreamReader streamReader = Util.newXMLInputFactory().createXMLStreamReader(bomInputStream);
```

Comment:
Can we use the simplest/shortest code below that we can?  I think the complexity below is hiding the purpose of above.

For example, do we need the `XMLStreamReader` instead of an `InputStreamReader`?

Reply:
Done.